### DISCUSSION TITLE
v10.0.0

### DISCUSSION THREAD

**Discussion by frankie567** [fastapi-users/fastapi-users#976] [General]
## Breaking changes

Version 10 marks important changes in how we manage User models and their ID.

Before, we were relying only on Pydantic models to work with users. In particular the [`current_user` dependency](https://fastapi-users.github.io/fastapi-users/10.0/usage/current-user/) would return you an instance of `UserDB`, a Pydantic model. This proved to be quite problematic with some ORM if you ever needed to **retrieve relationship data** or make specific requests.

Now, FastAPI Users is designed to always return you a **native object for your ORM model**, whether it's an SQLAlchemy model or a Beanie document. Pydantic models are now only used for validation and serialization inside the API.

Before, we were forcing the use of UUID as primary key ID; a consequence of the design above. This proved to be quite problematic on some databases, like MongoDB which uses a special ObjectID format by default. Some SQL folks also prefer to use traditional auto-increment integers.

Now, FastAPI Users is designed to use **generic ID type**. It means that you can use any type you want for your user's ID. By default, SQLAlchemy adapter still use UUID; but you can quite easily switch to another thing, like an integer. Beanie adapter for MongoDB will use native ObjectID by default, but it also can be overriden.

## Documentation improvements

* From this day, the documentation is **versioned**. It means that you'll be able read the documentation for older versions, starting at [v9.3](https://fastapi-users.github.io/fastapi-users/9.3/). There is a menu switch on top! 

<hr /><em>This discussion was created from the release <a href='https://github.com/fastapi-users/fastapi-users/releases/tag/v10.0.0'>v10.0.0</a>.</em>

**Comment by LonelyVikingMichael** [comment]
I am very excited to try this out, I have a terrible subclass of `SQLAlchemyUserDatabase` used to handle relationships that I can now get rid of.

Thank you!

**Reply by frankie567** [reply]
Thank you for your feedback @LonelyVikingMichael 🙏 I'm glad it'll help you!

**Comment by caniko** [comment]
Sad to see `tortoise-orm` being dropped. The silver lining for me is that `sqlalchemy` seems to be more mature than the competition, and I think it was the right call to make the switch.

I think more work on the `fastapi-user` side is needed for better integration with the capabilities of `sqlalchemy`, at least on this one issue that I have migrating from `tortoise-orm`. Please read at my [idea](https://github.com/fastapi-users/fastapi-users/discussions/977) for more information :pray:
